core/msg_bus.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_msg
+ *
+ * @{
+ *
+ * @file
+ * @brief       Messaging Bus API for inter process message broadcast.
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include "irq.h"
+#include "msg_bus.h"
+
+void msg_bus_init(msb_bus_t *bus)
+{
+    static uint16_t bus_count;
+
+    bus->subs.next = NULL;
+    bus->id = bus_count++;
+}
```

Comment:
there's one issue now: if threads create one-shot busses often, the count will run out (and be re-used). We need to document that busses should be long-running.

Reply:
I added a note about it and documented the maximum number of busses.

Reply:
:+1: